docs/TpchBenchmark.md

```diff
@@ -0,0 +1,94 @@
+# I/O Optimizations and the TpchBenchmark
+## Introduction
+This document is the outcome from a cycle of benchmarking to determine the best I/O performance against AWS S3 Parquet data for TPCH. It is intended to describe the tuning process recommendations and general suggestions for tuning the Velox query engine.
+
+Benchmarking in Velox is made easy with the optionally built TpchBenchmark (velox_tpch_benchmark) executable. To build the benchmark executable _(\_build/release/velox/benchamrks/tpch/velox_tpch_benchmark)_, use the following command line to do the build with S3 support:
+```bash
+$ make release EXTRA_CMAKE_FLAGS="-DVELOX_BUILD_BENCHMARKS=ON -DVELOX_ENABLE_S3=ON"
+```
+
+---
+## Use Cases for Velox
+Velox is a library so there are multiple ways to use it. This document will describe two models used by popular applications.
+
+### ___In-Process Multi-Threaded Executor Use Case___
+In this use case is used by [Presto](https://github.com/prestodb/presto) and, as it turns out, is the use case used by the [Velox TpchBenchmark Tool](#appendix-a-tpchbenchmark-tool-help-output). This use case uses a single multi-threaded process to perform execution of queries in parallel. Each query is broken up into threads called __drivers__ via a planning algorithm.  Each __driver__ may also have a thread pool to perform I/O in a parallel manner. In the benchmarking tool both __driver__ thread count, and I/O thread count are exposed as command line configuration options. In this use case, care must be taken to not create too many threads since maximum number of threads is a product of __driver__ threads and I/O threads. In this use case, the application owns creating __drivers__ and I/O Thread pools for Velox. In the case of the benchmark tool, the tool is responsible for both __driver__ threads and I/O threads. Presto likewise performs similar functions.
+
+### ___Multiple Process Executor Use Case___
+This use case is used by Spark + [Gluten](https://github.com/oap-project/gluten) and it differs from the Presto use case where parallelism is concerned. Spark uses multiple processes where each process is a Gluten+Velox query processor. Spark scales by using many Linux processes for query processing. In this case this means that the __drivers__ are outside of Velox and Gluten and is defined by the Spark configuration and number of workers. Gluten takes on the role of creating and exposing the I/O thread pool count to Spark as configuration and then injecting the I/O thread pool into Velox for parallel I/O.
```

Comment:
@zhouyuan - can you review this para on gluten?

Reply:
looks great!